src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState, useEffect, useRef } from 'react';
+import cn from 'classnames';
+import { EditField } from '../../types/types';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  onDelete?: (arg: number) => void;
+  onEdit?: (id: number, arg: EditField) => void;
+  isLoading?: boolean;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onDelete,
+  onEdit,
+  isLoading,
+}) => {
+  const { id, title, completed } = todo;
+  const [isEditable, setIsEditable] = useState(false);
+  const [value, setValue] = useState(title);
+  const inputEditRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (inputEditRef.current) {
+      inputEditRef.current.focus();
+    }
+  }, [isEditable]);
+
+  const handleClickOnTitle = () => {
+    setIsEditable(true);
+  };
+
+  const handleOnSubmit = () => {
+    const arg = {
+      title: value,
+    };
+
+    if (onEdit) {
+      onEdit(id, arg);
+    }
+  };
+
+  const handleCheck = () => {
+    const arg = {
```

Comment:
What is `arg`?
Choose a more meaningful name for the variable or pass the object to the function immediately